### PR TITLE
ref(quickstart): swap out Vagrant with Minikube

### PR DESCRIPTION
**Pull request by bacongobbler**
kube-up.sh has long been deprecated. This replaces the vagrant quickstart docs with minikube
instead as the successor for local kube development.

### CODE REVIEW

File: src/quickstart/provider/minikube/boot.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+* OS X
+    * [xhyve driver](https://github.com/kubernetes/minikube/blob/master/DRIVERS.md#xhyve-driver), [VirtualBox](https://www.virtualbox.org/wiki/Downloads) or [VMware Fusion](https://www.vmware.com/products/fusion) installation
+* Linux
+    * [VirtualBox](https://www.virtualbox.org/wiki/Downloads) or [KVM](http://www.linux-kvm.org/) installation
+* Windows
+    * [Hyper-V](https://github.com/kubernetes/minikube/blob/master/DRIVERS.md#hyperv-driver)
+* VT-x/AMD-v virtualization must be enabled in BIOS
+* The most recent version of `kubectl`. You can install kubectl following
+  [these steps](http://kubernetes.io/docs/getting-started-guides/kubectl/).
```
Comment (vdice) [inline src/quickstart/provider/minikube/boot.md]: nit: link redirects to https://kubernetes.io/docs/user-guide/prereqs/ so perhaps use this instead?

File: src/quickstart/provider/minikube/dns.md [hunk 2]
```diff
@@ -9,53 +9,53 @@ Kubernetes overlay network.
...
-The service address for the router is `10.247.114.249`.
+The service address for the router is `10.0.0.87`.
 
 Last, we need to inform your machine how to reach the service address.
 
 Add routes on Linux with:
+
 ```
-$ sudo route add 10.247.114.249 gw 10.245.1.3
+$ sudo route add 10.0.0.87 gw 192.168.99.100
```
Comment (mboersma) [inline src/quickstart/provider/minikube/dns.md]: Is adding routes really necessary? We suggest installing Workflow with `helm install --set router.host_port.enabled=true`, which should make things accessible at `$(minikube ip)`.

I just do: `$ deis register http://deis.$(minikube ip).nip.io`

Comment (bacongobbler) [inline src/quickstart/provider/minikube/dns.md]: Good point. I can remove this section. ping @krancour as I believe he contributed this.

File: src/quickstart/provider/minikube/boot.md (hunk 1 above)
Comment (bacongobbler) [inline src/quickstart/provider/minikube/boot.md]: 👍

File: src/quickstart/provider/minikube/dns.md (hunk 2 above)
Comment (slack) [inline src/quickstart/provider/minikube/dns.md]: All the route stuff can go, now that `minkube` is in the picture! Hooray!

**Pull request merged**